blocks/library/list/index.js

```diff
@@ -72,9 +53,8 @@ registerBlock( 'core/list', {
 		return (
 			<Editable
 				tagName={ nodeName.toLowerCase() }
-				onChange={ ( nextValues ) => {
-					setAttributes( { values: nextValues } );
-				} }
+				onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
+				onListChange={ ( listType ) => setAttributes( { nodeName: listType } ) }
```

Comment:
It's probably not going to be used by any other blocks but I don't mind changing the name.